downloader.py

```diff
@@ -0,0 +1,5 @@
+class Downloader:
+    def __init__(self, urls, timeout, max_attempts):
+        self.urls = urls
+        self.timeout = timeout
+        self.max_attempts = max_attempts
```

Comment:
Instance variable names should have an underscore prefix if they're not part of the public interface.

https://google.github.io/styleguide/pyguide.html#3162-naming-conventions